qiskit/qobj/pulse_qobj.py

```diff
@@ -0,0 +1,773 @@
+            self._data['memory_slot_size'] = memory_slot_size
+        if rep_time is not None:
+            self._data['rep_time'] = rep_time or []
+        if shots is not None:
+            self._data['shots'] = int(shots)
+
+        if max_credits is not None:
+            self._data['max_credits'] = int(max_credits)
+
+        if seed_simulator is not None:
+            self._data['seed_simulator'] = int(seed_simulator)
+
+        if memory_slots is not None:
+            self._data['memory_slots'] = int(memory_slots)
+
+        if kwargs:
+            self._data.update(kwargs)
+
+    @property
+    def meas_level(self):
+        """The measurement level to use."""
+        return self._data['meas_level']
+
+    @meas_level.setter
+    def meas_level(self, data):
+        self._data['meas_level'] = data
+
+    @property
+    def meas_return(self):
+        """The level of measurement information to return."""
+        return self._data['meas_return']
+
+    @meas_return.setter
+    def meas_return(self, data):
+        self._data['meas_return'] = data
+
+    @property
+    def shots(self):
+        """The number of shots."""
+        return self._data.get('shots')
+
+    @shots.setter
+    def shots(self, value):
+        self._data['shots'] = value
+
+    @property
+    def max_credits(self):
+        """The max_credits to use on the IBMQ public devices."""
+        return self._data.get('max_credits')
+
+    @max_credits.setter
+    def max_credits(self, value):
+        self._data['max_credits'] = value
+
+    @property
+    def memory_slots(self):
+        """The number of memory slots on the device."""
+        mem_slots = self._data.get('memory_slots')
+        if mem_slots is None:
+            raise AttributeError
+        return mem_slots
+
+    @memory_slots.setter
+    def memory_slots(self, value):
+        self._data['memory_slots'] = value
+
+    def to_dict(self):
+        """Return a dictionary format representation of the Pulse Qobj config.
+
+        Returns:
+            dict: The dictionary form of the PulseQobjConfig.
+        """
+        out_dict = copy.copy(self._data)
+        if hasattr(self, 'pulse_library'):
+            out_dict['pulse_library'] = [
+                x.to_dict() for x in self.pulse_library]
+
+        return out_dict
+
+    @classmethod
+    def from_dict(cls, data):
+        """Create a new PulseQobjConfig object from a dictionary.
+
+        Args:
+            data (dict): A dictionary for the config
+
+        Returns:
+            PulseQobjConfig: The object from the input dictionary.
+        """
+        if 'pulse_library' in data:
+            pulse_lib = data.pop('pulse_library')
+            pulse_lib_obj = [PulseLibraryItem.from_dict(x) for x in pulse_lib]
+            data['pulse_library'] = pulse_lib_obj
+        return cls(**data)
+
+    def __getattr__(self, name):
+        try:
+            return self._data[name]
+        except KeyError:
+            raise AttributeError
```

Comment:
Does this raise with the standard error message? Would be nice. Some of the other `AttributeErrors` probably need to be given error messages though

Reply:
That's a good point it won't have an exception message here. I'll add some more descriptive errors.

Reply:
Done in: https://github.com/Qiskit/qiskit-terra/pull/3383/commits/da6acb4bd8f655769bf9dbcfddce1ada8593c722